runtime_scan/pkg/orchestrator/scanconfigwatcher/watcher.go

```diff
@@ -0,0 +1,250 @@
+		scanConfigID, ok := scanConfig.GetID()
+		if !ok {
+			logger.Warnf("Skipping to invalid ScanConfig: ID is nil: %v", scanConfig)
+			continue
+		}
+
+		events = append(events, ScanConfigReconcileEvent{
+			ScanConfigID: scanConfigID,
+		})
+	}
+
+	return events, nil
+}
+
+// nolint:cyclop
+func (w *Watcher) Reconcile(ctx context.Context, event ScanConfigReconcileEvent) error {
+	logger := log.GetLoggerFromContextOrDiscard(ctx).WithFields(event.ToFields())
+	ctx = log.SetLoggerForContext(ctx, logger)
+
+	logger.Infof("Reconciling ScanConfig event")
+
+	scanConfig, err := w.backend.GetScanConfig(ctx, event.ScanConfigID, models.GetScanConfigsScanConfigIDParams{})
+	if err != nil || scanConfig == nil {
+		return fmt.Errorf("failed to fetch ScanConfig. Event=%s: %w", event, err)
+	}
+
+	_, ok := scanConfig.GetID()
+	if !ok {
+		return fmt.Errorf("invalid ScanConfig: ID is nil. Event=%s", event)
+	}
+
+	// nolint:gomnd
+	scheduleWindowSize := w.pollPeriod * 2
+	scheduleWindow := NewScheduleWindow(time.Now().UTC(), scheduleWindowSize)
+
+	scanConfigSchedule, err := NewScanConfigSchedule(scanConfig, scheduleWindow)
+	if err != nil {
+		return fmt.Errorf("failed to determine ScanConfig state: %w", err)
+	}
+
+	switch scanConfigSchedule.State {
+	case ScheduleStateDisabled:
+		logger.Debug("Skipping ScanConfig as it is disabled")
+	case ScheduleStateNotDue:
+		logger.Debugf("Skipping ScanConfig due to schedule: %s", scanConfigSchedule)
+	case ScheduleStateUnscheduled:
+		logger.Debug("Disable unscheduled ScanConfig")
+		if err = w.reconcileUnscheduled(ctx, scanConfig); err != nil {
+			return fmt.Errorf("failed to disable unscheduled ScanConfig: %w", err)
+		}
+	case ScheduleStateDue:
+		logger.Debug("Run new Scan for ScanConfig")
+		if err = w.reconcileDue(ctx, scanConfig, scanConfigSchedule); err != nil {
+			return fmt.Errorf("failed to create new Scan for ScanConfig: %w", err)
+		}
+	case ScheduleStateOverdue:
+		logger.Debug("Reschedule overdue ScanConfig")
+		if err = w.reconcileOverdue(ctx, scanConfig, scanConfigSchedule); err != nil {
+			return fmt.Errorf("failed to reschedule ScanConfig: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (w *Watcher) reconcileUnscheduled(ctx context.Context, scanConfig *models.ScanConfig) error {
+	scanConfigPatch := &models.ScanConfig{
+		Disabled: utils.PointerTo(true),
+	}
+
+	if err := w.backend.PatchScanConfig(ctx, *scanConfig.Id, scanConfigPatch); err != nil {
+		return fmt.Errorf("failed to patch ScanConfig. ScanConfigID=%s: %w", *scanConfig.Id, err)
+	}
+
+	return nil
+}
+
+func (w *Watcher) reconcileDue(ctx context.Context, scanConfig *models.ScanConfig, schedule *ScanConfigSchedule) error {
+	if err := w.createScan(ctx, scanConfig); err != nil {
+		return fmt.Errorf("failed to reconcile new Scan for ScanConfig. ScanConfigID=%s: %w", *scanConfig.Id, err)
+	}
+	nextOperationTime := schedule.OperationTime.NextAfter(schedule.Window.Next().Start())
+	// FIXME: disable ScanConfig if it was a oneshot
```

Comment:
what about this one?

Reply:
I deliberately skipped fixing this as it is non-critical and  it would require rerunning E2E manually to validate the change, therefore I'll address this in a separate PR.